torch_sparse/narrow.py

```diff
@@ -1,11 +1,69 @@
 from typing import Tuple
 
+import torch
 from torch_sparse.storage import SparseStorage
 from torch_sparse.tensor import SparseTensor
 
 
 def narrow(src: SparseTensor, dim: int, start: int,
-           length: int) -> SparseTensor:
+           length: int, bidim: False,
+           n_active_nodes: None) -> SparseTensor:
+    r"""It resizes :obj:`src` SparseTensor along one
+    specified dimension, or two dimensions at the same time
+    if the SparseTensor has 2 dimensions. In the
+    latter case, it returns a squared SparseTensor.
+    In the case of 1-dimensional resizing,
+    :obj:'start' and :obj:'lenght' are supported
+    In the case of bidimensional resizing, resizing
+    always starts from 0, and it returns a square matrix;
+    :obj:`src` must be a bidimensional SparseTensor
+    for bidim option to work.
+    Args:
+        src (SparseTensor): SparseTensor to be manipulated
+        dim (int): dimension along which the resizing is done.
+                    Not used in case bidim is True
+        start (int): first position to include in the ouput.
+                    Not used in case bidim is True (set to 0)
+        length (int): last position to include in the output.
+        bidim (bool):  it applies bidimensional and squared narrowing
+                    of the src, which must be a bidimensional SparseTensor.
+        n_active_nodes (int): Required only if bidim option is True;
+                    for efficient manipulation of the rowptr bidim case.
+    """
+    if bidim:
+        if len(src.storage._sparse_sizes) != 2:
+            raise NotImplementedError
+        
+        start = 0
+        if src.storage._rowptr is None:
+            rowptr, col, value = src.csr()
+
+        # rowptr
+        rowptr = torch.narrow(src.storage._rowptr,
+                              0, start, length + 1).clone()
+        rowptr[(n_active_nodes + 1):] = rowptr[n_active_nodes]
```

Comment:
The attempt is to keep the adj matrix squared. 
When we trim it on both dimensions, the last rows of the rowptr become full of zeros and to represent them, we extend the `rowptr `accordingly.  
If, however, having a non-square adj matrix is also fine in PyG, we can probably remove it, as far as I understand, and it would make the trim_to_layer even faster. 
Did I get your question right?

Reply:
My confusion is probably due to the interplay between `length` and `n_active_nodes`. Not exactly sure what's the difference.

Reply:
lenght is the size of the adj matrix. n_active _nodes is, at each layer, the number of nodes we want(need) to compute a representation for.